### PR TITLE
Update docs to mention that Glean is the way forward for telemetry

### PR DESCRIPTION
**Pull request by badboy**
---

### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### CODE REVIEW

File: components/service/glean/README.md [hunk 1]
```diff
@@ -1,9 +1,9 @@
 # [Android Components](../../../README.md) > Service > Glean
 
-A client-side telemetry SDK for collecting metrics and sending them to Mozilla's telemetry service
-(eventually replacing [service-telemetry](../telemetry/README.md)).
+A client-side telemetry SDK for collecting metrics and sending them to Mozilla's telemetry service.
+It is a replacment for the deprecated [service-telemetry](../telemetry/README.md).
```
Comment (brizental) [inline components/service/glean/README.md:4]:
```suggestion
It is a replacement for the deprecated [service-telemetry](../telemetry/README.md).
```

File: components/service/glean/README.md [hunk 2]
```diff
@@ -1,9 +1,9 @@
...
 
-Visit the [complete Glean documentation](https://mozilla.github.io/glean/). 
+Visit the [complete Glean documentation](https://mozilla.github.io/glean/).
```
Comment (mdboom) [inline components/service/glean/README.md:6]: This is the right link for this audience, but do we want to say "Glean SDK documentation" here?

Comment (badboy) [reply components/service/glean/README.md:6]: Sure, will update while I'm here anyway

Comment (badboy) [reply components/service/glean/README.md:6]:
```suggestion
Visit the [complete Glean SDK documentation](https://mozilla.github.io/glean/).
```

**Comment by badboy** [comment]
bors r=brizental

**Comment by badboy** [comment]
bors retry

that's an unrelated component that failed. I merely changed docs.

**Pull request merged**